workflow/models.py

```diff
@@ -1,63 +1,198 @@
     label = models.CharField(max_length=128, blank=False, null=False)
     description = models.TextField(null=False, blank=True)
-    workflow = models.ForeignKey('Workflow', related_name='sections', null=False)
     index = models.IntegerField(null=True, blank=True)
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='sections',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'sections'
+
     def __str__(self):
         return self.label
 
+
 class Widget(models.Model):
+    """Widget Model"""
+
     id = models.AutoField(primary_key=True)
     label = models.CharField(max_length=128, null=False, blank=False)
     description = models.TextField(null=False, blank=True)
     widget_type = models.CharField(max_length=24, blank=False, null=False)
-    parameter_mappings = models.ManyToManyField('WidgetParameterMapping', related_name='consumer_widgets')
-    section = models.ForeignKey('Section', related_name='widgets', null=False)
-    workflow = models.ForeignKey('Workflow', related_name='widgets', null=False)
     index = models.IntegerField(null=True, blank=True)
+
+    section = models.ForeignKey(
+        'Section',
+        related_name='widgets',
+        null=False
+    )
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='widgets',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'widgets'
+
     def __str__(self):
         return self.label
 
-class WidgetParameterMapping(models.Model):
+
+class ParameterAlias(models.Model):
+    """ParameterAlias Model"""
+
     id = models.AutoField(primary_key=True)
-    name = models.CharField(max_length=64, blank=False, null=False)
-    parameter = models.ForeignKey('Parameter', related_name='widget_parameter_mappings', null=True)
-    case = models.ForeignKey('Case', null=True)
-    workflow = models.ForeignKey('Workflow', related_name='widget_parameter_mappings', null=False)
+    alias = models.CharField(max_length=64, blank=False, null=False)
+
+    widget = models.ForeignKey(
+        'Widget',
+        related_name='parameter_aliases',
+        null=True
+    )
+
+    parameter_stub = models.ForeignKey(
+        'ParameterStub',
+        related_name='aliases',
+        null=True
+    )
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='parameter_aliases',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'parameter-aliases'
+
     def __str__(self):
-        return self.name
+        return self.alias
+
+
+class ParameterStub(models.Model):
+    """ParameterStub Model"""
 
-class Parameter(models.Model):
     id = models.AutoField(primary_key=True)
     name = models.CharField(max_length=64, blank=False)
-    value = JSONField()
-    properties = JSONField()
-    case = models.ForeignKey('Case', null=True)
-    workflow = models.ForeignKey('Workflow', related_name='parameters', null=False)
+    scope = models.CharField(
```

Comment:
Scope separates configuration of a workflow from the data a user actually enters. We don't use global, but it's there I guess.